### PR TITLE
UIOR-1211: Create "Bindery active" flag in POL and order templates

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UIOR-1211:](https://folio-org.atlassian.net/browse/UIOR-1211) Create "Bindery active" flag in POL and order templates
## Approach

https://github.com/folio-org/ui-orders/assets/116072773/dc09bc5a-bddb-4683-bf94-3f0e4e8435b8

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by Dmytro-Melnyshyn — APPROVED** [review]

File: src/common/POLFields/DetailsFields/FieldOrderFormat.js [hunk 1]
```diff
@@ -60,9 +62,12 @@ function FieldOrderFormat({
     });
   };
 
+  const isBindaryActive = get(formValues, 'details.isBindaryActive', false);
+  const dataOptions = isBindaryActive ? OPTIONS_WITH_BINDARY_ACTIVE : ORDER_FORMAT_OPTIONS;
```
Comment (usavkov-epam) [inline src/common/POLFields/DetailsFields/FieldOrderFormat.js:66]: What would a user see as the selected option if `electronicResource` or `other` was selected before activating `isBindaryActive`?

File: src/common/POLFields/utils.js [hunk 2]
```diff
@@ -10,3 +10,14 @@ export const isEresource = (format) => ERESOURCES.includes(format);
...
 export const isOtherResource = (format) => format === ORDER_FORMATS.other;
+
+export const isBindaryActiveDisabled = (orderFormat) => {
+  if (!orderFormat) {
+    return false;
+  }
+
+  return (
+    orderFormat !== ORDER_FORMATS.PEMix &&
+    orderFormat !== ORDER_FORMATS.physicalResource
```
Comment (usavkov-epam) [inline src/common/POLFields/utils.js:21]:
```suggestion
    orderFormat !== ORDER_FORMATS.PEMix
    && orderFormat !== ORDER_FORMATS.physicalResource
```

File: src/settings/InventoryRecordTypeSelectField.js [hunk 3]
```diff
@@ -3,16 +3,27 @@ import PropTypes from 'prop-types';
...
-const InventoryRecordTypeSelectField = ({ label, isRedux, ...rest }) => {
+const InventoryRecordTypeSelectField = ({
+  label,
+  disabled,
+  isRedux,
+  isBindaryActive = false,
+  ...rest
+}) => {
   const FieldComponent = isRedux ? FieldSelect : FieldSelectFinal;
+  const dataOptions = isBindaryActive ? OPTION_WITH_BINDARY_ACTIVE : INVENTORY_RECORDS_TYPE_FOR_SELECT;
```
Comment (usavkov-epam) [inline src/settings/InventoryRecordTypeSelectField.js:19]: What would a user see as the selected option if `instance`, `instanceAndHolding`, or `none` was selected before activating `isBindaryActive`?

Comment (usavkov-epam) [reply src/settings/InventoryRecordTypeSelectField.js:19]: That what I meant:

https://github.com/folio-org/ui-orders/assets/88109087/8f947c93-c4f1-4806-b3ba-ee803b0ee99c

Comment (alisher-epam) [reply src/settings/InventoryRecordTypeSelectField.js]: Thank you for the detailed review.


https://github.com/folio-org/ui-orders/assets/116072773/59aad65e-fc34-459e-8503-d2ee50c651d7

**Review by usavkov-epam — COMMENTED** [review]
Please do not pass `isBindaryActive` to field components and do not use `useEffect` to change the value inside the component. Field components should not modify themselves, this should happen from the outside. Pay attention to the examples when changing one field affects another one: `change` is called in the place that triggers the changing of the value of another field (search in github for `change(` [eg](https://github.com/search?q=repo%3Afolio-org%2Fui-orders%20change(&type=code))).

File: src/components/POLine/POLineForm.js [hunk 4]
```diff
@@ -245,6 +247,15 @@ function POLineForm({
     setHiddenFields(templateValue?.hiddenFields || {});
   }, [templateValue?.hiddenFields]);
 
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (alisher-epam) [reply src/components/POLine/POLineForm.js:250]: I see.

File: src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditor.js [hunk 5]
```diff
@@ -48,6 +51,7 @@ import {
   useFundDistributionValidation,
 } from '../../../common/hooks';
 import { omitFieldArraysAsyncErrors } from '../../../common/utils';
+import { OPTION_VALUE_WITH_BINDERY_ACTIVE } from '../../../components/POLine/const';
```
Comment (usavkov-epam) [inline src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditor.js:54]: code smell - unused

File: src/common/POLFields/DetailsFields/FieldBinderyActive.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    const { checked } = e.target;
+
+    if (checked) {
+      batch(() => {
+        change(FIELD_NAMES.isBinderyActive, checked);
+        change(FIELD_NAMES.orderFormat, OPTION_VALUE_WITH_BINDERY_ACTIVE);
+        change(FIELD_NAMES.receivingWorkflow, true);
+      });
+    } else {
+      change('details.bindery', false);
```
Comment (usavkov-epam) [inline src/common/POLFields/DetailsFields/FieldBinderyActive.js:31]: `FIELD_NAMES.isBinderyActive`?

**Review by usavkov-epam — APPROVED** [review]

**Pull request merged**